graph/metafeeds_test.go

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: MIT
+
+package graph
+
+import (
+	"fmt"
+
+	"github.com/ssb-ngi-pointer/go-metafeed"
+	"github.com/ssb-ngi-pointer/go-metafeed/metakeys"
+	"github.com/ssb-ngi-pointer/go-metafeed/metamngmt"
+	refs "go.mindeco.de/ssb-refs"
+)
+
+var metafeedsScenarios = []PeopleTestCase{
+	{
+		name: "metafeeds, simple",
+		ops: []PeopleOp{
+			PeopleOpNewPeerWithAglo{"alice", refs.RefAlgoFeedBendyButt},
+
+			PeopleOpNewSubFeed{
+				of:    "alice",
+				name:  "alice-legacy",
+				nonce: "test1",
+				algo:  refs.RefAlgoFeedSSB1,
+			},
+			PeopleOpNewSubFeed{
+				of:    "alice",
+				name:  "alice-gabby",
+				nonce: "test2",
+				algo:  refs.RefAlgoFeedGabby},
+		},
+		asserts: []PeopleAssertMaker{
+			PeopleAssertIsSubfeed("alice", "alice-legacy", true),
+			PeopleAssertIsSubfeed("alice", "alice-gabby", true),
+			PeopleAssertHops("alice", 0, "alice-legacy", "alice-gabby"),
+		},
+	},
+
+	{
+		name: "metafeeds of a friend",
+		ops: []PeopleOp{
+			PeopleOpNewPeer{"alice"},
+			PeopleOpNewPeerWithAglo{"bob", refs.RefAlgoFeedBendyButt},
+
+			// alice and bob are friends
+			PeopleOpFollow{"alice", "bob"},
+			PeopleOpFollow{"bob", "alice"},
+
+			// bobs subfeeds
+			PeopleOpNewSubFeed{of: "bob",
+				name:  "bob-legacy",
+				nonce: "test1",
+				algo:  refs.RefAlgoFeedSSB1},
+
+			PeopleOpNewSubFeed{of: "bob",
+				name:  "bob-gabby",
+				nonce: "test2",
+				algo:  refs.RefAlgoFeedGabby},
+		},
+		asserts: []PeopleAssertMaker{
+			PeopleAssertHops("alice", 0, "bob", "bob-legacy", "bob-gabby"),
+		},
+	},
+
+	{
+		name: "metafeed follows someone",
+		ops: []PeopleOp{
+			PeopleOpNewPeerWithAglo{"alice", refs.RefAlgoFeedBendyButt},
+			PeopleOpNewPeer{"some"},
+
+			PeopleOpNewSubFeed{
+				of:    "alice",
+				name:  "alice-legacy",
+				nonce: "test1",
+				algo:  refs.RefAlgoFeedSSB1,
+			},
+
+			PeopleOpFollow{"alice-legacy", "some"},
+		},
+		asserts: []PeopleAssertMaker{
+			PeopleAssertHops("alice", 0, "alice-legacy", "some"),
+		},
+	},
+}
+
+type PeopleOpNewSubFeed struct {
+	of, nonce, name string
+	algo            refs.RefAlgo
+}
+
+func (op PeopleOpNewSubFeed) Op(state *testState) error {
```

Comment:
each `Op` command could have an explanatory package comment one-liner. reading the subsequent code would be easier then, i think :)